frontend/src/shared/components/UnmountAnimation/UnmountAnimation.tsx

```diff
@@ -6,42 +6,49 @@ import {
 } from "react";
 import Polymorphic from "../Polymorphic/Polymorphic";
 
-interface DelayUnmountProps<T extends ElementType = "div"> {
+interface UnmountAfterAnimationProps<T extends ElementType = "div"> {
   visible: boolean;
   as?: T;
 }
 
-function UnmountAnimation({
+function UnmountAfterAnimation({
   visible,
   children,
   as = "div",
-}: PropsWithChildren<DelayUnmountProps>) {
+}: PropsWithChildren<UnmountAfterAnimationProps>) {
   const [mounted, setMounted] = useState(visible);
 
   useEffect(() => {
     if (visible) {
       setMounted(true);
-      return;
     }
   }, [visible]);
 
-  const unmount = () => {
-    if (visible) {
+  const refCallback = (element: HTMLElement | null) => {
+    if (!element || visible) {
       return;
     }
 
-    setMounted(false);
+    const animations = element.getAnimations({ subtree: true });
+    Promise.all(animations.map((animation) => animation.finished))
+      .then(() => {
+        setMounted(false);
+      })
+      .catch(() => {
+        // 애니메이션 중단 처리(abort error)
+        setMounted(false);
+      });
```

Comment:
```suggestion
    Promise.all(animations.map((animation) => animation.finished))
      .finally(() => {
        setMounted(false);
      });
```
현재 코드에서 finally로 대체해도 원하는대로 동작할 것 같은데 따로 나누신 이유는 의도를 명확히 하고 디버깅 등에 용이하게 하기 위해서 나누신걸까요?🤔